app/models/grant.rb

```diff
@@ -24,10 +27,31 @@ class Grant < ApplicationRecord
 
   enum state: GRANT_STATES
 
-  validates_presence_of :name
-  validates_presence_of :submission_open_date
-  validates_presence_of :submission_close_date
-  validates_presence_of :publish_date
+  before_destroy :deletable?
+  before_validation :prepare_slug, if: -> { slug.present? }
+
+  validates_presence_of :name,
+                        :slug,
+                        :submission_open_date,
+                        :submission_close_date,
+                        :publish_date
+
+
+  validates_uniqueness_of :slug, case_sensitive: false
+
+  validates      :name, uniqueness: true
+  validates      :slug,
+                 length: { minimum: 3,
+                           maximum: 15 },
```

Comment:
is this too short?

Reply:
I am not sure. I think you can set this however you want. I think that the there needs to be a check that the slug isn't just numbers or we might run into problems with it overlapping the record number in the future.